fighters/dedede/src/acmd/specials.rs

```diff
@@ -346,42 +407,188 @@ unsafe extern "C" fn game_specialairlwstart(agent: &mut L2CAgentBase) {
+        }
+        else{
+            ATTACK(agent, 0, 0, Hash40::new("hammer2"), 14.0 + (charge_level * 1.5), 361, 60, 0, 85, 8.0, 0.0, 0.0, 0.0, None, None, None, 1.0, 1.0, *ATTACK_SETOFF_KIND_OFF, *ATTACK_LR_CHECK_F, false, 10, 0.0, 0.0, false, false, false, false, true, *COLLISION_SITUATION_MASK_GA, *COLLISION_CATEGORY_MASK_ALL, *COLLISION_PART_MASK_ALL, false, Hash40::new("collision_attr_normal"), *ATTACK_SOUND_LEVEL_L, *COLLISION_SOUND_ATTR_DEDEDE, *ATTACK_REGION_HAMMER);
+            ATTACK(agent, 1, 0, Hash40::new("hammer1"), 14.0 + (charge_level * 1.5), 361, 60, 0, 85, 4.0, 10.0, 0.0, 0.0, None, None, None, 1.0, 1.0, *ATTACK_SETOFF_KIND_OFF, *ATTACK_LR_CHECK_F, false, 10, 0.0, 0.0, false, false, false, false, true, *COLLISION_SITUATION_MASK_GA, *COLLISION_CATEGORY_MASK_ALL, *COLLISION_PART_MASK_ALL, false, Hash40::new("collision_attr_normal"), *ATTACK_SOUND_LEVEL_L, *COLLISION_SOUND_ATTR_DEDEDE, *ATTACK_REGION_HAMMER);
+            ATTACK(agent, 2, 0, Hash40::new("hammer1"), 14.0 + (charge_level * 1.5), 361, 60, 0, 85, 4.0, 5.0, 0.0, 0.0, None, None, None, 1.0, 1.0, *ATTACK_SETOFF_KIND_OFF, *ATTACK_LR_CHECK_F, false, 10, 0.0, 0.0, false, false, false, false, true, *COLLISION_SITUATION_MASK_GA, *COLLISION_CATEGORY_MASK_ALL, *COLLISION_PART_MASK_ALL, false, Hash40::new("collision_attr_normal"), *ATTACK_SOUND_LEVEL_L, *COLLISION_SOUND_ATTR_DEDEDE, *ATTACK_REGION_HAMMER);
+            ATTACK(agent, 3, 0, Hash40::new("hammer1"), 14.0 + (charge_level * 1.5), 361, 60, 0, 85, 4.0, 0.0, 0.0, 0.0, None, None, None, 1.0, 1.0, *ATTACK_SETOFF_KIND_OFF, *ATTACK_LR_CHECK_F, false, 10, 0.0, 0.0, false, false, false, false, true, *COLLISION_SITUATION_MASK_GA, *COLLISION_CATEGORY_MASK_ALL, *COLLISION_PART_MASK_ALL, false, Hash40::new("collision_attr_normal"), *ATTACK_SOUND_LEVEL_L, *COLLISION_SOUND_ATTR_DEDEDE, *ATTACK_REGION_HAMMER);
+        }
+        
     }
-    frame(lua_state, 13.0);
+    frame(lua_state, 20.0);
     if is_excute(agent) {
-        AttackModule::clear_all(boma);
+        if !VarModule::is_flag(agent.battle_object, vars::dedede::instance::CONTINUE_JET_SPIN) && !agent.is_situation(*SITUATION_KIND_AIR){
+            KineticModule::add_speed(agent.module_accessor, &Vector3f{x: 0.6, y:0.0, z:0.0});
+        }
     }
-    frame(lua_state, 15.0);
+    frame(lua_state, 18.0);
+    if is_excute(agent){
+        VarModule::set_flag(agent.battle_object, vars::dedede::status::SPECIAL_LW_CONTINUE_SPIN, true);
+    }
+    frame(lua_state, 19.0);
     if is_excute(agent) {
-        FT_MOTION_RATE(agent, 43.0/(60.0-15.0));
+        MotionModule::set_rate(agent.module_accessor, 1.0);
     }
```

Comment:
avoid using `FT_MOTION_RATE` or similar motion-rating functions inside an `if is_excute(agent)` conditional. this applies to `MotionModule::set_rate` too!